WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift

```diff
@@ -29,6 +30,25 @@ final class ShippingLabelFormViewModel {
 
 // MARK: - Utils
 extension ShippingLabelFormViewModel {
+    // We generate the default origin address using the information
+    // of the logged Account and of the website.
+    private static func getDefaultOriginAddress() -> ShippingLabelAddress? {
+        // TODO: 2973 - for populating the origin address with the correct first and last name of the store owner
+        // we need to explicitly ask for first_name,last_name parameters in loadAccountSettings method
+        // and add the new properties to AccountSettings entity.
+        let address = Address(firstName: ServiceLocator.stores.sessionManager.defaultAccount?.displayName ?? "",
+                lastName: "", company: ServiceLocator.stores.sessionManager.defaultSite?.name ?? "",
+                address1: SiteAddress().address,
```

Comment:
This looks fine, but I noticed in testing that the address isn't always generated correctly. It looks like it's an issue with how we're actually setting `SiteAddress().address` here:

https://github.com/woocommerce/woocommerce-ios/blob/59ab55ba14463c84bc4f6b9cef0eea0b0e4963ab/WooCommerce/Classes/Tools/SiteAddress.swift#L53-L57

We're finding each value with `setting.settingID.contains(settingID)`, but looking for the key `woocommerce_store_address` can return the value for `woocommerce_store_address_2` instead, if it comes first. So we can end up with the wrong value set for `address1` here.

Reply:
Good catch Rachel! 💯 I substituted `containts` with `isEqual` 👍 0a7b4cce1dc2726e9261792d261af1e3f3e61b51